middle_end/flambda2/flambda2.ml

```diff
@@ -203,7 +203,16 @@ let lambda_to_cmm ~ppf_dump:ppf ~prefixname ~filename ~module_ident
               Flambda2_identifiers.Code_id.print (Code.code_id code)
               Cost_metrics.print size)
     end;
-    Flambda2_to_cmm.To_cmm.unit ~make_symbol:Compilenv.make_symbol flambda cmx
-      ~all_code
+    let cmm =
+      Flambda2_to_cmm.To_cmm.unit ~make_symbol:Compilenv.make_symbol flambda cmx
+        ~all_code
+    in
+    if not keep_symbol_tables then begin
```

Comment:
Why do we need to `keep_symbol_tables` sometimes? Why not resetting them unconditionally?

Reply:
Two cases where we need to keep them: being called from the native toplevel, so that the compiler is being invoked inside a REPL; and (as of the patch I just pushed) when more than one .ml file is passed on the command line